lib/puppet/defaults.rb

```diff
@@ -153,6 +153,13 @@ def self.default_diffargs
       :type       => :boolean,
       :desc       => "Whether to compile a static catalog."
     },
+    :static_catalogs_recursion => {
+      :default    => false,
+      :type       => :boolean,
+      :internal   => true,
```

Comment:
Added `:internal`, but we don't currently do anything or document what it means. I'm not sure where that should happen.